sympy/solvers/tests/test_ode.py

```diff
@@ -3768,3 +3768,17 @@ def test_2nd_2F1_hypergeometric():
           C2*hyper((S(1)/21, -S(11)/14), (S(5)/7,), x**(S(2)/7)))/(x**(S(2)/7) - 1)**(S(19)/84))
     assert sol == dsolve(eq, hint='2nd_hypergeometric')
     # assert checkodesol(eq, sol) == (True, 0) #issue-https://github.com/sympy/sympy/issues/17702
+
+
+def test_issue_15889():
+    sol = dsolve(exp(f(x).diff(x))-f(x)**2)
+    ans = Eq(Integral(1/log(y**2), (y, f(x))), C1 + x)
+    assert str(sol.as_dummy()) == str(ans.as_dummy())
```

Comment:
I wonder why `dummy_eq` doesn't work here...

Reply:
`dummy_eq` would have to use xreplace instead of subs since the physical dummy used is itself a logical dummy integration variable which can't be replaced with subs:
```
>>> y in Integral(x, (x, 1, 2)).subs(x, y).atoms()
False
```